### PR TITLE
Fix getting native query for mongo returns a broken string

### LINKED ISSUE #31122: Invalid native MongoDB query
### Describe the bug

This BE issue is related to FE issue #31121 - the extra quotes are added in FE.

Found as a part of code review discussion: https://github.com/metabase/metabase/pull/31111#discussion_r1209125938

image

### To Reproduce

1. Connect `QA Mongo4` (`mongo-4`) test database
2. Browse it
3. Browse `Orders` table
4. Click "Show editor" button
5. Click "View the native query" button
6. A modal opens. Inspect the generated query.

The generated query is not valid JSON i.e. it is not a valid Mongo query.
If you convert the question to a native query, the native query will fail to run.

### Expected behavior

Valid native Mongo query is generated

### Information about your Metabase installation

```
- Chrome Version 113.0.5672.126 (Official Build) (64-bit)
- Ubuntu 22.04
- Hash: edc969ce4247f810fe5fce806a19d77115ca80f1
```

### Severity

Looks like P1

### PR DESCRIPTION
**Pull request by qnkhuat**
Fixes  [Invalid native MongoDB query](https://github.com/metabase/metabase/issues/31122)

How to verify 
1. Connect to a non-empty mongo db
2. New question from any table
3. Click "View the native query"
4. it should show a json-like mongo query

### CODE REVIEW

File: test/metabase/db/query_test.clj [hunk 1]
```diff
@@ -76,24 +76,22 @@
...
-            weird-formatted-query (mdb.query/format-sql query :postgres)]
-        (testing "Formatting a non-sql query returns the same string"
-          (is (= query formatted-query)))
-        (testing "The wrong formatter will change the format..."
-          (is (not= query weird-formatted-query)))
-        (testing "...but the resulting data is still the same"
-          ;; Bottom line - Use the right driver, but if you use the wrong
-          ;; one it should be harmless but annoying
-          (is (= (json/parse-string query)
-                 (json/parse-string weird-formatted-query))))))))
```
Comment (calherries) [inline test/metabase/db/query_test.clj:96]: Why did you remove these tests?

Comment (markbastian) [reply test/metabase/db/query_test.clj:96]: The primary task for this test suite was to ensure we didn't get weird formatting as in `format-sql-with-params-test` below the above (See the expansion of { { } }). I updated the formatter for this. The mongo tests were there to capture behavior for if you happen to use the format-sql function on a mongo query string. I haven't looked at your changes, but that's the context.

Comment (qnkhuat) [reply test/metabase/db/query_test.clj:96]: yea but why do we need to handle that case? In which case we could pass the wrong driver? 

I can put that test back but I don't think allowing vagueness is a good practice.

Comment (qnkhuat) [reply test/metabase/db/query_test.clj]: Saw Mark is OOO and I want to lend this so I revert back those tests b3a96f29c3

File: src/metabase/db/query.clj [hunk 2]
```diff
@@ -35,25 +35,20 @@
 (set! *warn-on-reflection* true)
 
 (defn- format-sql*
-  "Return a nicely-formatted version of a generic `sql` string.
+  "Given a sql statement, return a nicely-formatted version of a generic `sql` string.
```
Comment (calherries) [inline src/metabase/db/query.clj:38]: I think it was fine how it was before. `sql` is implied as the argument
```suggestion
  "Return a nicely-formatted version of a generic `sql` string.
```

**Review by calherries — APPROVED** [review]
Looks good to me

**Pull request merged**